proxy/proxy.go

```diff
@@ -469,6 +469,9 @@ func setRequestURLForDynamicBackend(u *url.URL, stateBag map[string]interface{})
 
 func selectEndpoint(ctx *context, registry *routing.EndpointRegistry) *routing.LBEndpoint {
 	rt := ctx.route
+	if rt.BackendType != eskip.LBBackend {
+		return nil
+	}
```

Comment:
This way we have type check in two places.
Lets make it like it was before, i.e. selectEndpoint assumes correct type.